http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyRequestExecutor.java

```diff
@@ -176,9 +174,13 @@ private void makeRequestListener(Future<Channel> channelFuture) {
         if (channelFuture.isSuccess()) {
             channel = channelFuture.getNow();
             NettyUtils.doInEventLoop(channel.eventLoop(), () -> {
-                configureChannel();
-                if (tryConfigurePipeline()) {
-                    makeRequest();
+                try {
+                    configureChannel();
+                    if (tryConfigurePipeline()) {
```

Comment:
Should we make `tryConfigurePipeline` throw directly? This would wrap all of its exceptions with `Failed to initiate request to...`. Or do you want to reserve that for less explicit failure cases?

Reply:
Sorry, I'm not following. 

I added this try/catch because exceptions raised within this runnable go unreported and the completable future returned to the customer never returns (likely causing a deadlock in the customer's code). This change could be in a separate commit, since it's unrelated/unneeded to this change.

I just added it here because I ran into this bug during implementation.

Reply:
`tryConfigurePipeline` raises several exceptions itself (by also invoking `handleFailure`). I was just wondering if we should wrap all of them with this new logic for the sake of consistency (by changing `tryConfigurePipeline` to simply `throw`). But perhaps that just unnecessarily nests the more pertinent error message.

Reply:
Ahh, I see. We can combine them, yeah. Good idea.